controller/src/beerocks/master/db/db.h

```diff
@@ -717,9 +717,21 @@ class db {
 
     std::shared_ptr<vaps_list_t> m_vap_list;
 
+    struct hash_fn {
```

Comment:
This definition should move to sMacAddr.h (and the name changed of course). (Note: also the operator== and operator!= definitions should move from net_struct.h to sMacAddr.h - in a separate commit of course).

Also, I think it could be defined as the standard hash function for sMacAddr.

```
template <> struct std::hash<sMacAddress> {
    size_t operator()(const sMacAddr &m) const
    ...
};
```

You might need to play around with the namespaces to get this right.

Reply:
I actually thought like you but sMacAddr.h is an auto-generated file, when I do dist-clean all my changes there will be gone, so how to do this right? edit the YAML file? @arnout

Reply:
Ah, you're right, I thought sMacAddr.h was just copied from the src directory.

Indeed, then it is not possible to move it to sMacAddr.h. So instead it should move to net_struct.h.

And the rename to `std::hash` should also be possible of course.

Reply:
I planed to move it to net_struct.h but I see now that you already merged this PR.
if I want to add this do you want me to open another PR or revert this one? @arnout

Reply:
Nevermind I can see you have done this already, thanks 👍